### DISCUSSION TITLE
MGCB Editor will not open due to incomplete MonoGame extension installation

### DISCUSSION THREAD

**Discussion by northsnowbear** [MonoGame/MonoGame#8468] [General]
I teach game programming in a public school using MonoGame in a lab of Windows PCs with Visual Studio 2022 Community installed. I have students who can install the MonoGame extension the first time and run it perfectly. However, if the student moves to a new computer in the same classroom and installs the MonoGame extension the following folders will not be included. 

C:\Users\username\\.nuget\packages\dotnet-mgcb
C:\Users\username\\.nuget\packages\dotnet-mgcb-editor
C:\Users\username\\.nuget\packages\dotnet-mgcb-editor-linux
C:\Users\username\\.nuget\packages\dotnet-mgcb-editor-mac
C:\Users\username\\.nuget\packages\dotnet-mgcb-editor-windows

This results in the MGCB Editor not opening when the Content.mgcb file is clicked. 

The manual workaround is to copy the missing folders into place from another user profile. I have to manually associate the Content.mgcb file with the MGCB Editor executable file. So I can get the MGCB Editor to open. However, the MGCB Editor can no longer find the templates like SpriteFont.spritefont.

Does anybody know why the subsequent installations are failing to copy these folders or how to fix it?

**Comment by stromkos** [comment]
With the move to Nuget and SDK style projects, there is no longer a machine based install, it is all user-based.


The easiest solution is to follow the installation instructions for each student on all computers they will be using.

The actual solution:
---

Delete any files you may have copied. The metadata will claim the Nuget Package is already installed.

Open a command prompt (run CMD.exe, not powershell) `cd` to an existing project folder.  Run `dotnet restore`.

It will install all of the dependencies for that user.

**OR**

Open an existing MonoGame project under the new user.  Run it once. 

You may still need to manually associate the `.mgcb` file type in the editor, but all templates should be available.

The `dotnet restore` , performed during the run above, will install MonoGame  and the mgcb editor as a dependency in the user cache, including the template files and registry settings.
